Candidate: Andrés Felipe Bolaños Acosta <br>

Position: Business Intelligence Developer<br>

E-mail: afb.acosta@gmail.com<br>

Mobile phone: +5511973662765<br>

In order to execute SQL commands to query information from the database’s table two functions were implemented “connect” and “search query”. The first one allows us to connect directly to the mySQL data base and the second one allow us to make queries:

In [7]:
import mysql.connector

def connect():
    connection = (mysql.connector.connect(
                                  user = 'looqbox-challenge', 
                                  password = 'looq-challenge',
                                  host = '35.199.127.241',
                                  database = 'looqbox_challenge'))
    return connection

def search_query(query, conection_database):
    cursor = conection_database.cursor()
    cursor.execute(query)
    items_query = cursor.fetchall()
    
    for item in items_query:
        print(item)
        
    cursor.close()

In [8]:
conection_database = connect()

# What are the 10 most expensive products in the company?

In [9]:
query = ("SELECT PRODUCT_NAME FROM data_product "
         "ORDER BY PRODUCT_VAL DESC LIMIT 10 ")
search_query(query, conection_database)

('Whisky Escoces THE MACALLAN Ruby Garrafa 700ml com Caixa',)
('Whisky Escoces JOHNNIE WALKER Blue Label Garrafa 750ml',)
('Cafeteira Expresso 3 CORACOES Tres Modo Vermelho',)
('Vinho Portugues Tinto Vintage QUINTA DO CRASTO Garrafa 750ml',)
('Escova Dental Eletrica ORAL B D34 Professional Care 5000 110v',)
('Champagne Rose VEUVE CLICQUOT PONSARDIM Garrafa 750ml',)
('Champagne Frances Brut Imperial MOET Rose Garrafa 750ml',)
('Conjunto de Panelas Allegra em Inox TRAMONTINA 5 Pecas Gratis Utensilios 5 Pecas',)
('Whisky Escoces CHIVAS REGAL 18 Anos Garrafa 750ml',)
('Champagne Frances Brut Imperial MOET & CHANDON Garrafa 750ml',)


# What sections do the 'BEBIDAS' and 'PADARIA' departments have?:


In [10]:
query = ("SELECT DISTINCT DEP_NAME, SECTION_NAME "
         "FROM data_product WHERE DEP_NAME = 'BEBIDAS' "
         "OR DEP_NAME = 'PADARIA'ORDER BY DEP_NAME ")
search_query(query, conection_database)

('BEBIDAS', 'BEBIDAS')
('BEBIDAS', 'CERVEJAS')
('BEBIDAS', 'REFRESCOS')
('BEBIDAS', 'VINHOS')
('PADARIA', 'DOCES-E-SOBREMESAS')
('PADARIA', 'GESTANTE')
('PADARIA', 'PADARIA')
('PADARIA', 'QUEIJOS-E-FRIOS')


The department 'BEBIDAS' has sections 'BEBIDAS', 'CERVEJAS', 'REFRESCOS' and 'VINHOS'. While, "DOCES-E-SOBREMESAS", "GESTANTE", "PADARIA" and "QUEIJOS-E-FRIOS" correspond to sections of 'PADARIA'

# What was the total sale of products (in dollars) of each Business Area in the first quarter of 2019?


In [11]:
query = ("SELECT A.BUSINESS_NAME, SUM(B.SALES_VALUE) "
         "FROM data_store_cad AS A "
         "INNER JOIN data_product_sales AS B "
         "ON A.STORE_CODE = B.STORE_CODE "
         "WHERE B.DATE >= '2019-01-01' AND B.DATE <= '2019-03-31' "
         "GROUP BY A.BUSINESS_NAME")
search_query(query, conection_database)

('Atacado', Decimal('80384884.60'))
('Farma', Decimal('81776691.73'))
('Posto', Decimal('32072326.40'))
('Proximidade', Decimal('80171122.80'))
('Varejo', Decimal('81032347.65'))


Making the previous result clear:
   
   ATACADO:  80,384,884.60
   
   FARMA: 81,776,691.73
   
   POSTO: 32,072,326.40
   
   PROXIMIDADE: 80,171,122.80
   
   VAREJO: 81,032,347.65
   

The same values are also achieved if we apply another "INNER JOIN" to the tables "data_store_cad" and "data_store_sales":

In [12]:
query = ("SELECT A.BUSINESS_NAME, SUM(B.SALES_VALUE) "
         "FROM data_store_cad AS A "
         "INNER JOIN data_store_sales AS B "
         "ON A.STORE_CODE = B.STORE_CODE "
         "WHERE B.DATE >= '2019-01-01' AND B.DATE <= '2019-03-31' "
         "GROUP BY A.BUSINESS_NAME ")
search_query(query, conection_database)

('Atacado', 80384884.6)
('Farma', 81776691.73)
('Posto', 32072326.4)
('Proximidade', 80171122.8)
('Varejo', 81032347.65)


It should be mentioned that “Atacado”, “Farma”, “Posto”, “Proximidade” and “Varejo” are all the business areas in the table “data_store_cad”. However, a particular case occurs when any of those business areas did not have sales in the first quarter of 2019, just for that case a “LEFT JOIN”  (“data_store_cad” being the left table) is appropriate to show that any of those areas did not generated income (but this is not the case).
  